### PR TITLE
10 model edit page

### PR DESCRIPTION
**Pull request by olfarmer**

### CODE REVIEW

File: frontend/src/scene/ClickObjects.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    const matched = state.raycaster.intersectObjects(state.scene.children);
+
+
+    if (matched.length > 0) {
+        const first = matched[0];
+
+        // Highlight
+        first.object.scale.set(1.2, 1.2, 1.2);
+
+        console.log("Clicked " + first.object.name);
```
[the remark this answers is not in the document]
Comment (12rcu) [reply frontend/src/scene/ClickObjects.tsx:22]: also `console.debug` is usually more helpful the .log()

File: frontend/src/scene/Rotate.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+    state.scene.traverse((node: Object3D) => {
+        if (node instanceof Light) {
+            node.intensity *= 0.25; // Scale down light intensity
+        }
+
+        if (Object.keys(node.userData).length > 0 && node.userData["prop"]) {
+            sel.current = node;
+            console.log(`Custom properties for ${node.name}:`, node.userData);
```
Comment (12rcu) [inline frontend/src/scene/Rotate.tsx:18]: remove debug?

File: frontend/src/scene/ClickObjects.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import {RootState, useThree} from "@react-three/fiber";
+
+
+function ClickObjects() {
+    const state = useThree();
+    const canvas = state.gl.domElement;
+    canvas.addEventListener('click', () => OnClick(state));
+
+    return null;
```
[the remark this answers is not in the document]
Comment (olfarmer) [reply frontend/src/scene/ClickObjects.tsx:9]: Return null in a React Component indicates that there is nothing to render in this component. After some research, also [return false](https://tech.jotform.com/return-null-vs-return-false-in-react-826d8abcc429) could be used.

File: frontend/src/scene/Rotate.tsx (hunk 2 above)
Comment (olfarmer) [reply frontend/src/scene/Rotate.tsx]: Done

File: frontend/src/scene/Edit.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    const fileInputRef = useRef<HTMLInputElement | null>(null);
+    const [gltfUrl, setGltfUrl] = useState<string | null>(null);
+
+    const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const file = event.target.files?.[0];
+
+        if (file) {
+            // Create an object URL from the file
+            const objectUrl = URL.createObjectURL(file);
+            console.log("Generated Object URL:", objectUrl);
```
[the remark this answers is not in the document]
Comment (olfarmer) [reply frontend/src/scene/Edit.tsx]: Done

File: frontend/src/scene/ClickObjects.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (olfarmer) [reply frontend/src/scene/ClickObjects.tsx]: Done

**Pull request merged**